src/database/migration/v2.js

```diff
@@ -188,6 +132,61 @@ const dataMigrations = [
       });
     },
   },
+  {
+    version: '2.3.6',
+    migrate: database => {
+      // Find all the transactions with no item batches, if any
+      const transactionBatches = database
+        .objects('TransactionBatch')
+        .filtered('itemBatch == $0', null)
+        .snapshot();
+      // Do nothing if there aren't any.
+      if (!(transactionBatches.length > 0)) return;
+      // For each of these item batchless-transaction batches, create a new
+      // 0 quantity item batch
+      transactionBatches.forEach(transactionBatch => {
+        if (!transactionBatch) return;
+        const {
+          itemId,
+          batch,
+          expiryDate,
+          packSize,
+          costPrice,
+          sellPrice,
+          donor,
+        } = transactionBatch;
+        // TransactionBatch are not directly related, just hold a reference. Find
+        // the actual Item record.
+        const items = database.objects('Item').filtered('id == $0', itemId);
+        // Double checking the itemId in an ItemBatch is actually related to an item.
+        // Early exit if not, something else is wrong
+        if (!items.length > 0) return;
+        const item = items[0];
+        // Create the new ItemBatch using as much detail from the TransactionBatch
+        // as possible
+        database.write(() => {
+          const newItemBatch = database.update('ItemBatch', {
+            id: generateUUID(),
+            batch,
+            expiryDate,
+            packSize,
+            costPrice,
+            sellPrice,
+            donor,
+            item,
+          });
+          // Add the transaction batch into the new item batch
+          newItemBatch.transactionBatches.push(transactionBatch);
+          database.save('ItemBatch', newItemBatch);
+          // Update the TransactionBatch with the newly created ItemBatch
+          database.update('TransactionBatch', {
+            ...transactionBatch,
```

Comment:
for some reason i've never been a fan of full destructure in update, would rather just id. (if anything to be changed in this PR it's this)